### PR TITLE
mat-1121 add simulation for staking module

### PR DESCRIPTION
**Pull request by ptsayli**
Package: app
coverage: 49.1% of statements

### CODE REVIEW

File: app/sim_test.go [hunk 1]
```diff
@@ -204,22 +202,20 @@ func TestAppSimulationAfterImport(t *testing.T) {
...
-		newApp.ModuleAccountAddrs(), config,
-	)
-	require.NoError(t, err)
+	require.Panics(t, func() {
+		simulation.SimulateFromSeed(
+			t, os.Stdout, newApp.BaseApp, AppStateFn(app.Codec(), app.SimulationManager()),
+			SimulationOperations(newApp, newApp.Codec(), config),
+			newApp.ModuleAccountAddrs(), config,
+		)
+	})
```
Comment (jdkanani) [inline app/sim_test.go:218]: This shouldn't panic!

Comment (ptsayli) [reply app/sim_test.go:218]: We have set block limit` to 500, and that is already imported from `previous State`
so it panic here
```
prevHeight := app.LastBlockHeight()
if req.Header.Height != prevHeight+1 {
	return fmt.Errorf("invalid height: %d; expected: %d", req.Header.Height, prevHeight+1)
}
```

Comment (jdkanani) [reply app/sim_test.go:218]: Maybe we are using old DB. Block should start from 0 after import, but the init state should be old.

File: app/sim_test.go [hunk 2]
```diff
@@ -204,22 +202,20 @@ func TestAppSimulationAfterImport(t *testing.T) {
 		require.NoError(t, os.RemoveAll(newDir))
 	}()
 
-	newApp := NewHeimdallApp(logger, db, nil)
-	if err := newApp.LoadLatestVersion(newApp.keys[bam.MainStoreKey]); err != nil {
-		require.NoError(t, err)
-	}
+	newApp := NewHeimdallApp(logger, db)
```
Comment (jdkanani) [inline app/sim_test.go:205]:
```suggestion
	newApp := NewHeimdallApp(logger, newDB)
```

**Pull request merged**